### DISCUSSION TITLE
How to avoid events go through dialog window to underlying window?

### DISCUSSION THREAD

**Discussion by ruimo** [emilk/egui#1585] [Q&A] [answered]
Hi, I am trying to show small dialog using egui::Window(). But events on my dialog go through underlying main window too. How to avoid this? I prepared a small example tweaking hello_world example:

https://github.com/ruimo/egui-dialog-test.git

It shows a small dialog on top of main window. When you move mouse cursor, it prints "mouse move" events that reach to underlying main window. It prints events even if the mouse cursor is on the dialog.

**Comment by emilk** [comment, marked as answer]
Clicks shouldn't go through windows, `ui.input()` is just short for `ui.ctx().input()`, and so shows all input to all of egui.

If you want to know if the pointer is in a `Ui`, use https://docs.rs/egui/latest/egui/struct.Ui.html#method.ui_contains_pointer

**Reply by ruimo** [reply]
Thanks! Using ui_contains_pointer() has one difficulty for me. If I drag an object in the main window to out of the window, event will not be fired any more. I want to treat this gesture to request auto scrolling the contents in the main window. I noticed I can prepare a flag that reflects the dialog is showing and stop handling events according to the flag.
